cmd/utils.go

```diff
@@ -607,6 +608,23 @@ func newCustomHTTPTransport(tlsConfig *tls.Config, dialTimeout time.Duration) fu
 	}
 }
 
+// NewGatewayHTTPTransportWithClientCerts returns a new http configuration
+// used while communicating with the cloud backends.
+func NewGatewayHTTPTransportWithClientCerts(clientCert, clientKey string) *http.Transport {
+	transport := newGatewayHTTPTransport(1 * time.Minute)
+	if clientCert != "" && clientKey != "" {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+		defer cancel()
+		c, err := certs.NewManager(ctx, clientCert, clientKey, tls.LoadX509KeyPair)
+		if err != nil {
+			logger.LogIf(ctx, fmt.Errorf("failed to load client key and cert, please check your endpoint configuration: %s",
+				err.Error()))
+		}
+		transport.TLSClientConfig.GetClientCertificate = c.GetClientCertificate
```

Comment:
Upon error `c` can be empty leading to a crash here., avoid that by not setting the GetClientCertificate()